pytorch_lightning/callbacks/device_stats_monitor.py

```diff
@@ -43,33 +52,62 @@ class DeviceStatsMonitor(Callback):
         >>> trainer = Trainer(callbacks=[device_stats]) # doctest: +SKIP
     """
 
-    def setup(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", stage: Optional[str] = None) -> None:
-        if not trainer.loggers:
-            raise MisconfigurationException("Cannot use DeviceStatsMonitor callback with Trainer that has no logger.")
+    def __init__(self, cpu_stats: Optional[bool] = None) -> None:
+        self.cpu_stats = cpu_stats
 
-    def on_train_batch_start(
+    def setup(
```

Comment:
IMO yes, in case the user modifies the trainer in between.

Reply:
what do you mean modifies? It doesn't log anything if it's not training.

Reply:
If I do:
```py
trainer = Trainer(callbacks=[DeviceStatsMonitor()], logger=False)
trainer.validate(model)
```
IMO it should raise any exception since DeviceStatsMonitor doesn't do anything during `validate/test/predict`.

Reply:
Mmm true. We should add support for those in the future

Reply:
then let's update this too later? @kaushikb11